### DISCUSSION TITLE
Shell api

### DISCUSSION THREAD

**Discussion by Reaviik** [cc-tweaked/CC-Tweaked#1402] [General]
Estou criando um "spotify" com a api Shell usando o comando Shell.run("speaker play link") já convertido, porem não encontrei em nenhum lugar uma forma de alterar o volume do speaker, me pergunto se é possível

**Comment by piprett** [comment]
https://tweaked.cc/peripheral/speaker.html#v:playAudio
speaker.playAudio(audio, volume)

**Reply by SquidDev** [reply]
To expand on this - there's no way to pass a volume to the `speaker` program. Instead you'll need to write your own version of it using the [`dfpwm` parser](https://tweaked.cc/library/cc.audio.dfpwm.html#v:make_decoder) and [`speaker.playAudio`](https://tweaked.cc/peripheral/speaker.html#v:playAudio)
